runtime/gc_vlhgc/ConfigurationIncrementalGenerational.cpp

```diff
@@ -132,6 +132,20 @@ MM_ConfigurationIncrementalGenerational::createHeapWithManager(MM_EnvironmentBas
 	}
 #endif /* defined(OMR_GC_VLHGC_CONCURRENT_COPY_FORWARD) */
 
+#if defined(J9VM_GC_ENABLE_DOUBLE_MAP)
+	/* Disable double map if requested page sizes used is equal to huge pages.
+	 * Currently double map is not supported when huge pages are used.
+	 */
+	if (NULL != heap) {
+		if (extensions->memoryManager->isLargePage(env, heap->getPageSize())) {
+			GC_ArrayObjectModel *indexableObjectModel = &extensions->indexableObjectModel;
+			if (indexableObjectModel->isDoubleMappingEnabled()) {
```

Comment:
May sound overkill, but I'm thinking to extend this logic a bit more....

Let's introduce _config_ param in Extension called doubleMappingRequested (currently false by default, but soon to change for supported platforms). This is what command line option would blindly control.

And then we have the existing _status_ param in indexableObjectModel where the info is if we obeyed the request or not. Currently there is only one reason to not obey, but perhaps there will be more (well, I hope there will be less - which makes this overkill change).

Here at early startup instead of querying if status is already set, we query if it's requested. The rest of the code (mid run) would query the status flag.

The control flag could reside in ExtensionsVLHGC, hence no need to touch OMR.

Reply:
I'd move the code up just after MM_HeapVirtualMemory::newInstance() (if that's the spot where we determine effective page size)